GliaWidgets/ViewModel/EngagementViewModel.swift

```diff
@@ -54,10 +56,14 @@ class EngagementViewModel {
         self.alertConfiguration = alertConfiguration
         self.screenShareHandler = screenShareHandler
         interactor.addObserver(self, handler: interactorEvent)
+        screenShareHandler.status.addObserver(self) { status, _ in
+            self.onScreenSharingStatusChange(status)
```

Comment:
Let's use weak reference to self to avoid memory leak.

Reply:
Ah, the same mistake again. Thanks

Reply:
Fixed this in other places too